File: app/src/main/java/com/sofps/inspirationalquotes/asynctask/ScreenshotLoader.java
package com.sofps.inspirationalquotes.asynctask;

import android.content.Context;
import android.graphics.Bitmap;
import android.os.AsyncTask;
import com.sofps.inspirationalquotes.util.ScreenshotUtils;
import java.io.File;
import java.util.UUID;

public class ScreenshotLoader extends AsyncTask<Bitmap, Void, File> {

    public interface ScreenshotLoaderTaskListener {

        void onScreenshotLoaderTaskComplete(File file);

        void onScreenshotLoaderTaskInProgress();
    }

    private final Context mContext;
    private final ScreenshotLoaderTaskListener mListener;

    public ScreenshotLoader(Context context, ScreenshotLoaderTaskListener listener) {
        mContext = context;
        mListener = listener;
    }

    @Override
    protected File doInBackground(Bitmap... bitmaps) {
        File saveFile = ScreenshotUtils.getMainDirectoryName(mContext); // Get the path to save screenshot
        String filename = "IQ_" + UUID.randomUUID() + ".jpg";
        return ScreenshotUtils.store(bitmaps[0], filename, saveFile); // Save the screenshot to selected path
    }

    @Override
    protected void onPreExecute() {
        mListener.onScreenshotLoaderTaskInProgress();
    }

    @Override
    protected void onPostExecute(File result) {
        mListener.onScreenshotLoaderTaskComplete(result);
    }
}

File: app/src/main/java/com/sofps/inspirationalquotes/asynctask/ScreenshotLoader.kt
package com.sofps.inspirationalquotes.asynctask

import android.content.Context
import android.graphics.Bitmap
import android.os.AsyncTask
import com.sofps.inspirationalquotes.util.ScreenshotUtils
import java.io.File
import java.util.UUID

class ScreenshotLoader(
        private val context: Context,
        private val listener: ScreenshotLoaderTaskListener
) : AsyncTask<Bitmap, Void, File>() {

    interface ScreenshotLoaderTaskListener {

        fun onScreenshotLoaderTaskComplete(file: File)

        fun onScreenshotLoaderTaskInProgress()
    }

    override fun doInBackground(vararg bitmaps: Bitmap): File {
        // Get the path to save screenshot
        val saveFile = ScreenshotUtils.getMainDirectoryName(context)
        val filename = "IQ_" + UUID.randomUUID() + ".jpg"
        // Save the screenshot to selected path
        return ScreenshotUtils.store(bitmaps[0], filename, saveFile)
    }

    override fun onPreExecute() {
        listener.onScreenshotLoaderTaskInProgress()
    }

    override fun onPostExecute(result: File) {
        listener.onScreenshotLoaderTaskComplete(result)
    }
}
